torch/_dynamo/utils.py

```diff
@@ -774,6 +774,40 @@ def torch_clone(x):
             # Access data_ptr() for a xla tensor will cause crash
             return torch_clone(x)
 
+        # Handle sparse storage (no stride).
+        if x.layout is torch.sparse_coo:
+            return torch.sparse_coo_tensor(
+                torch_clone(x._indices()), torch_clone(x._values()), x.shape
+            )
+        elif x.layout is torch.sparse_csr:
+            return torch.sparse_csr_tensor(
+                torch_clone(x.crow_indices()),
+                torch_clone(x.col_indices()),
+                torch_clone(x.values()),
+                x.shape
+            )
```

Comment:
Let's reduce LOC here, say, via
```suggestion
        elif is_sparse_compressed(x):
            if x.layout in {torch.sparse_csr, torch.sparse_bsr}:
                 compressed_indices = x.crow_indices()
                 plain_indices = x.col_indices()
            else:
                 compressed_indices = x.ccol_indices()
                 plain_indices = x.row_indices()
            return torch.sparse_compressed_tensor(
                torch_clone(compressed_indices),
                torch_clone(plain_indices),
                torch_clone(x.values()),
                x.shape,
                layout=x.layout
            )
```
where
```python
from torch._subclasses.meta_utils import is_sparse_compressed
```

Reply:
oh, I love it! very good suggestion, thanks!